merge percona xtrabackup plugin into master (#386)
bareos/bareos: docs/manuals/source/TasksAndConcepts/Plugins.rst

.. _PerconaXtrabackupPlugin:
.. _backup-mysql-XtraBackup:

Backup of MySQL Databases using the Bareos MySQL Percona XtraBackup Plugin
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

:index:`\ <single: Plugin; MySQL Backup>`
:index:`\ <single: Percona XtraBackup>`
:index:`\ <single: XtraBackup>`

This plugin uses Perconas XtraBackup tool, to make full and incremental backups of Mysql / MariaDB databases.

The key features of XtraBackup are:

- Incremental backups
- Backups that complete quickly and reliably
- Uninterrupted transaction processing during backups
- Savings on disk space and network bandwidth
- Higher uptime due to faster restore time

Incremental backups only work for INNODB tables, when using MYISAM, only full backups can be created.

Prerequisites
'''''''''''''

Install the XtraBackup tool from Percona. Documentation and packages are available here: https://www.percona.com/software/mysql-database/percona-XtraBackup. The plugin was successfully tested with XtraBackup versions 2.3.5 and 2.4.4.

As it is a Python plugin, it will also require to have the package **bareos-filedaemon-python-plugin** installed on the |fd|, where you run it.

> For authentication the :file:`.mycnf` file of the user running the |fd|. Before proceeding, make sure that XtraBackup can connect to the database and create backups.

Reviewer: ```suggestion
??? For authentication the :file:`.mycnf` file of the user running the |fd| ???. Before proceeding, make sure that XtraBackup can connect to the database and create backups.
```

Reviewer: Den Satz verstehe ich nicht.

Installation
''''''''''''

> Make sure you have met the prerequisites. Install the package **bareos-filedaemon-percona_XtraBackup-python-plugin**.

Reviewer: ```suggestion
Make sure you have met the prerequisites, after that install the package **bareos-filedaemon-percona_XtraBackup-python-plugin**.
```

:file:`/tmp/bareos-restores/_percona/`

Each restore job gets an own subdirectory, because Percona expects an empty directory. In that subdirectory, a new directory is created for every backup job that was part of the Full-Incremental sequence.

The naming scheme is: :file:`fromLSN_toLSN_jobid`

Example:

::

   /tmp/bareos-restores/_percona/351/
   |-- 00000000000000000000_00000000000010129154_0000000334
   |-- 00000000000010129154_00000000000010142295_0000000335
   |-- 00000000000010142295_00000000000010201260_0000000338

This example shows the restore tree for restore job with ID 351. First subdirectory has all files from the first full backup job with ID 334. It starts at LSN 0 and goes until LSN 10129154.

Next line is the first incremental job with ID 335, starting at LSN 10129154 until 10142295. The third line is the 2nd incremental job with ID 338.

To further prepare the restored files, use the :command:`XtraBackup --prepare` command. Read https://www.percona.com/doc/percona-xtrabackup/2.4/backup_scenarios/incremental_backup.html for more information.

Troubleshooting
'''''''''''''''
If things don't work as expected, make sure that

> - Bareos FileDaemon (FD) works in general, so that you can make simple file

Reviewer: ```suggestion
- Check that the |fd| (FD) works in general, so that you can make simple file
```

>   backups  and restores - Bareos FD Python plugins work in general, try one of

Reviewer: ```suggestion
  backups  and restores - check that the Bareos FD Python plugins work in general, try one of
```

>   - the Bareos FD Python plugins work in general, try one of

Author: ```suggestion
- the Bareos FD Python plugins work in general, try one of
```
